samples/boards/up_squared/gpio_counter/README.rst

```diff
@@ -0,0 +1,95 @@
+.. _up_squared gpio_counter:
+
+UP Squared
+##########
+
+Overview
+********
+This sample provides an example of how to configure GPIO input and output to
+the UP Squared board.
+
+The sample enables a pin as GPIO input (active high) that triggers the increment
+of a counter (range is 0 to size of counter_pins[]). The counter increments for
+each change from 0 to 1 on HAT Pin 16 (BIOS Pin 19). The value of the counter is represented
+on GPIO output (active high) as a 4-bit value
```

Comment:
While I think the intent was for the pins to be a 4-bit representation of the counter value, i.e., the pins go 0000, 0001, 0010, 0011, 0100, ..., 1111, it looks like the code masks the counter with 0b11 (for 4 pins) so there's an error in the code. Or I could be wrong...

Reply:
Thanks for asking.  I do not think there is an issue with that aspect (there are other known issues).

I tested the existing sample just now (cherry-picked 005aff7), and it behaves as 'intended' - the output pins display the count through a value of 0x0 to 0xf displayed in binary across the pins.  For example, 0xc (serial console) is represented by 0b1100. Scaling the sample to less or more output pins would be neat exercise...so, I need to update the range described here.